kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
```

Comment:
> Googled around a bit and it seems the query parameter is the better option

Could you share those sources? Curious to see.

The way I see this, those two endpoints do not really give you the same entities, or at least, not in the same **context**, which is everything here:

- `/upstreams/{upstream}/targets` gets you a - paginated - list of *all* the target rows currently in the **database**.
- `/upstream/{upstream}/target/active/` gets you a subset of those above rows, currently in use in the **nodes memory**, at *runtime*.

It's all about the context in which those entities are presented here. I think it is significant enough to warrant another endpoint. Additionally, we now have a single endpoint that is paginated by default, but not paginated when the `?active=` qs param is given. Overall, I think it's fine not to paginate active targets, but I do think this is another reason why it shouldn't be under the same endpoint.